src/coreclr/jit/assertionprop.cpp

```diff
@@ -68,6 +68,50 @@ bool IntegralRange::Contains(int64_t value) const
     return SymbolicToRealMap[static_cast<int32_t>(value)];
 }
 
+//------------------------------------------------------------------------
+// GetUpperBound: Get the largest possible value "type" can represent
+//
+// Arguments:
+//    node     - the node, of an integral type, in question
+//    type     - the integral type in question
+//    compiler - the Compiler, used to retrieve additional info
+//    
+// Return Value:
+//    the largest possible value "type" can represent.
+//
+/* static */ size_t IntegralRange::GetUpperBound(GenTree* node, var_types type, Compiler* compiler)
+{
+    IntegralRange valRange = IntegralRange::ForNode(node, compiler);
+#if defined(HOST_X86) || defined(HOST_ARM)
+    return (int32_t)IntegralRange::SymbolicToRealValue(valRange.UpperBoundForType(type));
+#else
+    return IntegralRange::SymbolicToRealValue(valRange.UpperBoundForType(type));
+#endif
+
+}
+
+//------------------------------------------------------------------------
+// GetUpperBound: Get the smallest possible value "type" can represent
+//
+// Arguments:
+//    node     - the node, of an integral type, in question
+//    type     - the integral type in question
+//    compiler - the Compiler, used to retrieve additional info
+//
+// Return Value:
+//    the smallest possible value "type" can represent.
+//
+/* static */ size_t IntegralRange::GetLowerBound(GenTree* node, var_types type, Compiler* compiler)
+{
+    IntegralRange valRange = IntegralRange::ForNode(node, compiler);
+#if defined(HOST_X86) || defined(HOST_ARM)
+    return (int32_t)IntegralRange::SymbolicToRealValue(valRange.LowerBoundForType(type));
+#else
+    return IntegralRange::SymbolicToRealValue(valRange.LowerBoundForType(type));
+
+#endif
+}
```

Comment:
This is not what I meant. You should add simple instance methods on `IntegralRange` that just return `m_lowerBound` and `m_upperBound` (two fields on that class). You should not use `LowerBoundForType` or `UpperBoundForType` at all for this logic (those are static methods that you are calling through an instance).

In compiler.h, just add the following:
```diff
diff --git a/src/coreclr/jit/compiler.h b/src/coreclr/jit/compiler.h
index 37aa9d06350..082f9ddacda 100644
--- a/src/coreclr/jit/compiler.h
+++ b/src/coreclr/jit/compiler.h
@@ -1235,20 +1235,30 @@ private:
 
 public:
     IntegralRange() = default;
 
     IntegralRange(SymbolicIntegerValue lowerBound, SymbolicIntegerValue upperBound)
         : m_lowerBound(lowerBound), m_upperBound(upperBound)
     {
         assert(lowerBound <= upperBound);
     }
 
+    SymbolicIntegerValue GetLowerBound()
+    {
+        return m_lowerBound;
+    }
+
+    SymbolicIntegerValue GetUpperBound()
+    {
+        return m_upperBound;
+    }
+
     bool Contains(int64_t value) const;
 
     bool Contains(IntegralRange other) const
     {
         return (m_lowerBound <= other.m_lowerBound) && (other.m_upperBound <= m_upperBound);
     }
 
     bool IsPositive()
     {
         return m_lowerBound >= SymbolicIntegerValue::Zero;
```

Then call `IntegralRange::ForNode` from your new transform and use these getters to get the lower and upper bound of the range.